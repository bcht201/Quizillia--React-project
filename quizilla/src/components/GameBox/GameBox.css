.questionContainer{
    display: flex;
    align-items: center;
    flex-direction: column;
    padding: 0 60px;
}

.answerButton{
    min-width: 50px;
    margin-bottom: 15px;
    padding: 4px;
}

h3 {
    margin-top: 2%;
    margin-bottom: 30px;
    padding: 0;
}

.buttonContainer{
    display: flex;
    flex-flow: wrap;
    flex-direction: row;
}