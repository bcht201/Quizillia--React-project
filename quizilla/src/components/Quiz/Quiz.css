.currentPlayer {
    text-align: center;
    margin-top: 5%;
    font-size: 30px;
    color: green;
}