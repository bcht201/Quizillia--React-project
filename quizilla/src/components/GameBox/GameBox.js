import React from 'react'
import {Parser} from 'html-to-react';  
import './GameBox.css';

const GameBox = (props) => {
    const formatQuestion = () => {
        let question = props.gameInfo.question
        const htmlToReactParser = new Parser();
        const reactElement = htmlToReactParser.parse(question)
        return reactElement
    }

    let answers = [...props.gameInfo.incorrect_answers, props.gameInfo.correct_answer].sort();  
    const onClickHandler = (event) => {
        props.calculateScore(event.target.value);
    }

    return(
        <div className = "questionContainer">
            <h3>{formatQuestion()}</h3>
            {answers.map(answer => {
                const htmlToReactParser2 = new Parser();
                return <div className = "buttonContainer"> 
                            <button className = "answerButton" 
                                value = {answer} 
                                id = {answer} 
                                onClick={(event)=> onClickHandler(event)}>{htmlToReactParser2.parse(answer)}
                            </button>
                        </div>
            })}
        </div>
    )
}

export default GameBox