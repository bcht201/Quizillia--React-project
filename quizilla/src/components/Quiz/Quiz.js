import React from 'react'
import axios from 'axios'
import GameBox from '../GameBox/GameBox';

class Quiz extends React.Component{
    constructor(props) {
        super(props);
        this.state= {
            data: [],
            index: 0,
            score: 0
        }
    }

    componentDidMount() {
        this.apiCall();
    }



    apiCall = () => {
        axios.get(`https://opentdb.com/api.php?amount=${this.props.maxNumberOfQuestions}&category=${this.props.catID}&difficulty=${this.props.diff}`)
        .then(response => {
            console.log(response.data.results)
            this.setState({data: response.data.results})})
    }

    calculateScore = (answer) =>{
        console.log("I calculated the score");
        let points;
        if(answer === this.state.data[this.state.index].correct_answer){
            points = 1;
        } else{
            points = -1;
        }
        this.setState(prevState => {return ({score: prevState.score + points, index: prevState.index + 1})})
        
    }

    generateQuestion = () => {
        if(this.state.index === this.props.maxNumberOfQuestions) {
           let emojis = [<p>&#128563;</p>, <p>&#128513;</p> ];
            return(<p> You got {this.state.score}/{this.props.maxNumberOfQuestions} {this.state.score < 0? emojis[0] : emojis[1]} </p>)
        }
        else if(this.state.data.length !== 0) { 
            return( <GameBox gameInfo={this.state.data[this.state.index]} 
            calculateScore = {this.calculateScore}/>) 
        }
        else {
            return;
       }
    }

    render() {
        return (
            <div>
                {this.generateQuestion()}
            </div>
            
        )
    }
}

export default Quiz;