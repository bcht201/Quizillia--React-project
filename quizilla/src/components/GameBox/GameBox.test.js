import React from 'react';
import {
        shallow,
        } from 'enzyme';
import GameBox from './GameBox';
import Quiz from './../Quiz/Quiz';
import renderer from 'react-test-renderer'

describe('GameBox.js', () => {
    let wrapper;
    let mockData;

    beforeEach(() => {
        mockData = {"category": "Mythology",
        "type": "multiple",
        "difficulty": "hard",
        "question": "Talos, the mythical giant bronze man, was the protector of which island?",
        "correct_answer": "Crete",
        "incorrect_answers": ["Sardinia",
        "Sicily",
        "Cyprus"
        ]
        }
       
       
        wrapper= shallow(<GameBox gameInfo = {mockData}  />);
    })

    it('should match snapshot', () => {
        const tree = renderer.create(<GameBox gameInfo = {mockData} />).toJSON();
        expect(tree).toMatchSnapshot();
    })


    //p tag should not render without being called
    it('to have 1 container <div /> and 1 for each answer button', () => {
        expect(wrapper.find('div').length).toEqual(5);
    })

    it('should have 4 answer buttons', () => {
        expect(wrapper.find('button').length).toEqual(4);
    })

    it('should have buttons with answers as the value', () => {
        expect(wrapper.find('#Crete').text()).toEqual('Crete');
    })

    it('should render the question', ()=> {
       expect(wrapper.contains('<h3>Talos, the mythical giant bronze man, was the protector of which island?</h3>'));
    });
})

