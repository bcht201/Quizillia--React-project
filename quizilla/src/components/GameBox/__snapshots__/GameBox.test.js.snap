// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GameBox.js should match snapshot 1`] = `
<div
  className="questionContainer"
>
  <h3>
    Talos, the mythical giant bronze man, was the protector of which island?
  </h3>
  <div
    className="buttonContainer"
  >
    <button
      className="answerButton"
      id="Crete"
      onClick={[Function]}
      value="Crete"
    >
      Crete
    </button>
  </div>
  <div
    className="buttonContainer"
  >
    <button
      className="answerButton"
      id="Cyprus"
      onClick={[Function]}
      value="Cyprus"
    >
      Cyprus
    </button>
  </div>
  <div
    className="buttonContainer"
  >
    <button
      className="answerButton"
      id="Sardinia"
      onClick={[Function]}
      value="Sardinia"
    >
      Sardinia
    </button>
  </div>
  <div
    className="buttonContainer"
  >
    <button
      className="answerButton"
      id="Sicily"
      onClick={[Function]}
      value="Sicily"
    >
      Sicily
    </button>
  </div>
</div>
`;
